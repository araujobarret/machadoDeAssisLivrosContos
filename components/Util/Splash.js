import React from 'react';
import { Dimensions, View, Image, StyleSheet } from 'react-native';


class Splash extends React.Component {
  static navigationOptions = {
    headerStyle: { height: 0 }
  };

  startLoading() {
    setTimeout(() => {
      this.props.loadingStop();
    }, 1000);
  }

  // The main question about the logo is the image size
  // The image needs to fit a flex ratio provided, so the image needs to be nested in two
  // containers to be not too large or too tiny
  render() {
    return (
      <View style={SplashStyles.container}>
        <Image
          style={SplashStyles.imageLoading}
          resizeMode="cover"
          onLoad={() => this.startLoading()}
          source={require('../../assets/imgs/loading.jpg')}
        />
      </View>
    );
  }
}

const deviceWidth = Dimensions.get("window").width;
const deviceHeight = Dimensions.get("window").height;

const SplashStyles = {
  container: {
    flex: 1,
  },
  imageLoading: {
    width: deviceWidth,
    height: deviceHeight,
  },
};


export default Splash;
