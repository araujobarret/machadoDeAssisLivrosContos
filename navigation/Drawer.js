import React from 'react';
import { StyleSheet, Text, View, Image, ScrollView, TouchableOpacity } from 'react-native';
import { DrawerItems, SafeAreaView, NavigationActions } from 'react-navigation';

function translateItem (label) {
  switch (label) {
    case 'Home': return 'LIVROS';
    case 'Search': return 'BUSCAR';
    case 'About': return 'SOBRE';
  }
}

function _renderDrawerItem (navigation) {
  const items = [];
  const keys = Object.keys(navigation.router.childRouters.Main.childRouters);
  for (let i = 0; i < keys.length; i++) {
    // BookNavigator don't need to be in the menu
    if (keys[i] == 'BookChapters') { continue; }
    if(keys[i] !== 'BookNavigator') {
      if (navigation.state.routes[0].routes[navigation.state.routes[0].index].routeName === keys[i]) {
        items.push(
          <TouchableOpacity key={'menuItem_' + i}
            style={styles.menuItem}
            onPress={() => navigation.dispatch(navigation.actions.closeDrawer())}
          >
            <Text style={[styles.label, styles.activeLabel]}>{ translateItem(keys[i]) }</Text>
          </TouchableOpacity>
        );
      } else {
        items.push(
          <TouchableOpacity key={'menuItem_' + i}
            style={styles.menuItem}
            onPress={() => navigation.dispatch(NavigationActions.navigate({ routeName: keys[i] }))}
          >
            <Text style={styles.label}>{ translateItem(keys[i]) }</Text>
          </TouchableOpacity>
        );
      }
    }
  }
  return items;
}

export const Drawer = (props) => {
  return (
    <ScrollView>
      <SafeAreaView style={styles.container} forceInset={{ top: 'always' }}>
        <Text style={styles.title}>MENU</Text>
        { _renderDrawerItem (props.navigation) }
      </SafeAreaView>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  title: {
    textAlign: 'center',
    fontSize: 22,
    fontWeight: 'bold',
    marginTop: 16,
    marginBottom: 24,
  },
  menuItem: {
    marginLeft: 24,
    height: 60,
    justifyContent: 'center'
  },
  label: {
    color: '#333333',
    fontSize: 18,
    fontWeight: 'bold',
  },
  activeLabel: {
    color: '#e91e63'
  }
});
