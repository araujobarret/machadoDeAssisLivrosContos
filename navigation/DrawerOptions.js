import React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';

class DrawerOptions extends React.Component {
  render() {
    const { navigation } = this.props
    return (
      <View>
        <Text>Drawer options</Text>
      </View>
    )
  }
}


export default DrawerOptions;
