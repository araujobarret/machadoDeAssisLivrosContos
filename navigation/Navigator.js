import React from 'react';
import { Button, Image, Platform, ScrollView, View, TouchableOpacity } from 'react-native';
import { createStackNavigator, createDrawerNavigator, SafeAreaView, DrawerItems, createAppContainer } from 'react-navigation';

import { DrawerActions, StackActions, NavigationActions } from 'react-navigation';
import { Drawer } from './Drawer';
import Splash from '../components/Util/Splash';
import Home from '../components/Home/Home';
import BookNavigator from '../components/Book/BookNavigator';
import Search from '../components/Search/Search';

const MainStackNavigator = createStackNavigator(
  {
    Home: { screen: Home },
    BookNavigator: { screen: BookNavigator },
    Search: { screen: Search }
  },
  {
    initialRouteName: 'Home',
    defaultNavigationOptions: ({ navigation }) => ({
      headerStyle: {
        backgroundColor: '#f4511e',
      },
      headerTintColor: '#fff',
      headerTitleStyle: {
        fontWeight: 'bold',
        textAlign: 'center',
        flex: 1
      },
      headerLeft: (
        <TouchableOpacity
          style={{ marginLeft: 10, paddingVertical: 10 }}
          onPress={() => navigation.dispatch(DrawerActions.openDrawer())}
        >
          <Image source={require('../assets/imgs/menu.png')} resizeMode="contain" style={{ width: 24 }}/>
        </TouchableOpacity>
      ),
      headerRight: (
        <View style={{ marginRight: 10 }} />
      )
    })
  }
);

const SearchStackNavigator = createStackNavigator(
  {
    Search: { screen: Search }
  },
  {
    initialRouteName: 'Search',
    defaultNavigationOptions: ({ navigation }) => ({
      headerStyle: {
        backgroundColor: '#fff',
        elevation: 0,
      },
      headerTintColor: '#000',
      headerTitleStyle: {
        fontWeight: 'bold',
        textAlign: 'center',
        flex: 1
      },
      headerLeft: (
        <TouchableOpacity
          style={{ marginLeft: 10, paddingVertical: 10 }}
          onPress={() => {
            // Reset the Search component and then go back to home screen
            navigation.dispatch(
              StackActions.reset({
                index: 0,
                actions: [ NavigationActions.navigate({ routeName: 'Search' })]
              })
            );
            navigation.dispatch(NavigationActions.back());
          }}
        >
          <Image source={require('../assets/imgs/cancel.png')} resizeMode="contain" style={{ width: 36 }}/>
        </TouchableOpacity>
      ),
      headerRight: (
        <View style={{ marginRight: 10 }} />
      )
    })
  }
);

const MainDrawerNavigator = createDrawerNavigator(
  {
    Livros: { path: '/', screen: MainStackNavigator },
    Buscar: { path: '/search', screen: SearchStackNavigator}
  },
  {
    initialRouteName: 'Livros',
    contentComponent: (props) => <Drawer {...props} />,
    contentOptions: {
      activeTintColor: '#e91e63',
      itemsContainerStyle: {
        paddingTop: 24
      },
      itemStyle: {
        justifyContent: 'center',
      },
      labelStyle: {
        fontSize: 20
      },
      iconContainerStyle: {
        opacity: 1
      }
    }
  }
);

// const defaultGetStateForActions = MainDrawerNavigator.router.getStateForAction;
//
// MainDrawerNavigator.router.getStateForAction = (action, state) => {
//   console.log('action', action);
//   console.log('state', state);
//   return defaultGetStateForActions(action, state);
// };

export const Navigator = createAppContainer(MainDrawerNavigator);
