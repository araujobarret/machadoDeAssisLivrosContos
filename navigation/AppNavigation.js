import React, { Component } from "react";
import { BackHandler, Platform } from "react-native";
import { Navigator } from "./Navigator";

// Component to control the initialization of the stack navigator
class AppNavigation extends Component {

  componentWillMount(){
      // Method to handle android back button
      BackHandler.addEventListener('hardwareBackPress', function(){
        let {dispatch, navigation, nav} = this.props;
        if(nav.routes.length === 1 && nav.routes[0].routeName === "Root"){
          return false;
        }
        dispatch({ type: "Navigation/BACK"});
        return true;
      }.bind(this));
  }

  componentWillUnmount(){
    BackHandler.removeEventListener("hardwareBackPress");
  }

  render() {
    return (
      <Navigator />
    );
  }
}


export default AppNavigation;
