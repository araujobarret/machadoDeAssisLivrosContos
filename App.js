import React from 'react';
import { View } from 'react-native';
import { Provider } from 'react-redux';
import "unorm";

import { Navigator } from './navigation/Navigator';
import Splash from './components/Util/Splash';
import configureStore from './store/store';

const store = configureStore();

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isLoading: true };
  }

  render() {
    return (
      <Provider store={store}>
        { this.state.isLoading ? <Splash loadingStop={() => this.setState({ isLoading: false })}/> : <Navigator /> }
      </Provider>
    );
  }
}
