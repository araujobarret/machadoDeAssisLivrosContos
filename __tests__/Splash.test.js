import React from 'react';
import { Image } from 'react-native';
// setup file
import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from '../App';
import { Navigator } from '../navigation/Navigator';
import Splash from '../components/Util/Splash';
import utils from '../utils';

configure({ adapter: new Adapter() });

describe('Testing Splash component', () => {
  const wrapper = shallow(<Splash />);

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
