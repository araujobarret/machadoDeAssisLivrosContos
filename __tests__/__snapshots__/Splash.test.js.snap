// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Splash component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Image
        onLoad={[Function]}
        resizeMode="cover"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
      />,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "onLoad": [Function],
        "resizeMode": "cover",
        "source": Object {
          "process": [Function],
        },
        "style": Object {
          "height": 1334,
          "width": 750,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Image
          onLoad={[Function]}
          resizeMode="cover"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 1334,
              "width": 750,
            }
          }
        />,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onLoad": [Function],
          "resizeMode": "cover",
          "source": Object {
            "process": [Function],
          },
          "style": Object {
            "height": 1334,
            "width": 750,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
